prism/src/main/java/org/apache/falcon/resource/proxy/ExtensionManagerProxy.java

```diff
@@ -586,6 +597,10 @@ public APIResult update(
         String extensionName = getExtensionName(jobName);
         try {
             entityMap = getEntityList(extensionName, jobName, feedForms, processForms, config);
+            if (entityMap.isEmpty()) {
+                // return failure if the extension job doesn't exist
```

Comment:
entityMap won't be empty. But the list of values in them can be empty. One of them can be empty. Both may not be empty. Can have a check on the same for both not being empty.